packages/core/test/combinator/loop-test.js

```diff
@@ -24,8 +24,8 @@ describe('loop', function () {
 
     return collectEventsFor(a.length, s)
       .then(events => {
-        assert.same(a.length, events.length)
-        assert.equals([
+        is(a.length, events.length)
```

Comment:
is() works fine here, but eq() is probably the right fit conceptually.  In general, I tend to think of eq() as the right thing for primitive values.

I feel like a good rule of thumb is:  Use eq() in all cases unless your intent is to test that two "pointers" point to the same object.